File: HashMaps/M3/M3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        longest = 0
        tracking = set()
        for i in range(len(s)):
            while s[i] in tracking:
                tracking.discard(s[left])
                left += 1
            tracking.add(s[i])
            longest = max(longest, i-left+1)
        return longest

File: HashMaps/M3/M3.ts
/*
Longest Substring Without Repeating Characters
Given a string s, find the length of the longest
substring
 without repeating characters.
*/

function lengthOfLongestSubstring(s: string): number {
	let left = 0;
	let longest = 0;
	const substring = new Set();
	for (let i = 0; i < s.length; i++) {
		while (substring.has(s[i])) {
			substring.delete(s[left]);
			left++;
		}
		substring.add(s[i]);
		longest = Math.max(longest, i - left + 1);
	}
	return longest;
}

//! 9min 21sec
